javascript/modules/client/src/client.ts

```diff
@@ -282,32 +308,124 @@ export class Client extends ClientCore implements IClient {
+  }
+
+  private async _getDaoBalances(
+    daoIdentifier: string
+  ): Promise<TokenBalance[]> {
+    // TODO: Implement actual fetch logic using subgraph.
+    // Note: it would be nice if the client could be instantiated with dao identifier
+
+    if (!daoIdentifier) {
+      throw new Error("A DAO identifier is needed");
+    }
+
+    const tokenBalances: TokenBalance[] = tokenList.map(
+      (token: TokenBalance["token"]) => ({
+        id: `${daoIdentifier}_${token.id}`,
+        token,
+        // Generate a random balance amount between [0, 1000]
+        balance: BigInt(
+          (Math.floor(Math.random() * (1000 - 1 + 1) + 1) * 10) ** 18
+        ),
+        // Generate a random date in the past
+        lastUpdated: Math.floor(
+          new Date(
+            +new Date() - Math.floor(Math.random() * 10000000000)
+          ).getTime() / 1000
+        ).toString(),
+      })
+    );
+
+    return Promise.resolve(tokenBalances);
+  }
+
+  private async _getDaoTransfers(
+    daoIdentifier: string
+  ): Promise<IDaoTransfers> {
+    // TODO: Implement actual fetch logic using subgraph.
+    // Note: it would be nice if the client could be instantiated with dao identifier
+
+    if (!daoIdentifier) {
+      throw new Error("A DAO identifier is needed");
+    }
+
+    // This is a temporary transfer list, needs to remove later
+    const transfers = [
+      {
+        sender: "0x9370ef1a59ad9cbaea30b92a6ae9dd82006c7ac0",
+        transaction:
+          "0x4c97c60f499dc69918b1b77ab7504eeacbd1e1a536e10471e12c184885dafc05",
+      },
+      {
+        sender: "0xb1dc5d0881eea99a61d28be66fc491aae2a13d6a",
+        transaction:
+          "0x6b0b8b815d78b83a5a69a883244a3ca2bdc25832edee2bc45e7b6392ad57fd94",
+      },
+      {
+        sender: "0x2db75d8404144cd5918815a44b8ac3f4db2a7faf",
+        transaction:
+          "0x08525a68b342be200c220f5a22d30425a262c5603e63c210b7664f26b8418bcc",
+      },
+      {
+        sender: "0x2db75d8404144cd5918815a44b8ac3f4db2a7faf",
+        transaction:
+          "0x8269a60f658e33393d3f20d065cd5107d410d41c5dba9e5c467efcd3f98db015",
+      },
+    ];
+
+    const vaultDeposits: DaoTransfer[] = tokenList.map(
+      (token: TokenBalance["token"], index: number) => ({
+        id: `${daoIdentifier}_${token.id}`,
+        dao: {
+          id: daoIdentifier,
+        },
+        token,
+        sender: transfers[index].sender,
+        // Generate a random amount between [0, 10]
+        amount: BigInt(
+          (Math.floor(Math.random() * (10 - 1 + 1) + 1) * 10) ** 18
+        ),
+        reference: "",
+        transaction: transfers[index].transaction,
+        // Generate a random date in the past
+        createdAt: Math.floor(
+          new Date(
+            +new Date() - Math.floor(Math.random() * 10000000000)
+          ).getTime() / 1000
+        ).toString(),
+      })
+    );
+
+    // Withdraw data structure would be similar to deposit list
+    const vaultWithdraws: DaoTransfer[] = [];
```

Comment:
Aren't we populating any data here?

Reply:
Yes, But for dummy data, I left an empty array.